### DISCUSSION TITLE
DAFoam for optimization in supersonic flight conditions

### DISCUSSION THREAD

**Discussion by usmanshahid90** [mdolab/dafoam#106] [General]
Can we use transonic dafoam for supersonic conditions? If yes, then what are the necessary changes and if not then how to do optimization in supersonic flight conditons?

**Comment by friedenhe** [comment]
DARhoSimpleCFoam did run for supersonic flow (M=1.2; see the following figure), but we have not verified the flow solutions or done any optimization for supersonic flow with DAFoam. It may work if the Mach number is a bit higher than one. Here is a case if you are interested. To try this case, unzip NACA0012_Supersonic.zip and run `python runScript.py`. Note that we have to use a good initial field, which was generated by using the upwind scheme. If you use a uniform flow field as the initial condition, the flow will blow up.

[NACA0012_Supersonic.zip](https://github.com/mdolab/dafoam/files/6232236/NACA0012_Supersonic.zip)
![pContour](https://user-images.githubusercontent.com/3891165/113057821-22708c80-9173-11eb-98f7-dda7666f4990.png)

**Reply by usmanshahid90** [reply]
Thank you very much for your reply. I  will look into it and will update you accordingly.

**Comment by 15175829310** [comment]
Have you ever done flow field inversion from 0.7 to 0.8 Mach? I am currently using an airfoil to do this work, but the gradient is not calculated.

**Comment by friedenhe** [comment]
What do you mean by "the gradient is not calculated"?
